crates/crypto/decaf377-fmd/src/precision.rs

```diff
@@ -0,0 +1,38 @@
+use core::fmt;
+
+use crate::Error;
+
+/// The maximum detection precision, chosen so that the message bits fit in 3 bytes.
+pub(crate) const MAX_PRECISION: u8 = 24;
+
+/// Represents the precision governing the false positive rate of detection.
+///
+/// This is usually measured in bits, where a precision of `n` bits yields false
+/// positives with a rate of `2^-n`.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct Precision(u8);
+
+impl Precision {
+    pub fn new(precision_bits: u8) -> Result<Self, Error> {
+        if precision_bits > MAX_PRECISION {
+            return Err(Error::PrecisionTooLarge(precision_bits));
+        }
+        Ok(Self(precision_bits))
+    }
```

Comment:
It'd be convenient to have `TryFrom` impls for numeric types so that it's possible to write
```
1.try_into().unwrap()
```
instead of forcing all consumers to import an extra `Precision` type they don't otherwise use